tekton-catalog/pipeline-loops/cmd/cli/main.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright 2021 The Kubeflow Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	pipelineloopv1alpha1 "github.com/kubeflow/kfp-tekton/tekton-catalog/pipeline-loops/pkg/apis/pipelineloop/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+func validateFlags(action, inputFileName, inputFileType string) {
+	var errs []error
+	if action != "validate" {
+		err := fmt.Errorf("unsupported action %s\n", action)
+		errs = append(errs, err)
+	}
+	if inputFileName == "" {
+		err := fmt.Errorf("missing input spec, please specify -f /path/to/file.(yaml|json)\n")
+		errs = append(errs, err)
+	}
+	if inputFileType != "yaml" && inputFileType != "json" {
+		err := fmt.Errorf("unsupported input file type:%s\n", inputFileType)
+		errs = append(errs, err)
+	}
+	if len(errs) > 0 {
+		fmt.Println("Errors while processing input:")
+		for i, e := range errs {
+			fmt.Printf("Error: %d -> %v\n", i, e)
+		}
+		os.Exit(1)
+	}
+}
+
+func main() {
+	var action, inputFileName, inputFileType string
+
+	flag.StringVar(&action, "a", "validate", "The `action` on the resource.")
+	flag.StringVar(&inputFileName, "f", "", "path to input `filename` either a yaml or json.")
+	flag.StringVar(&inputFileType, "file-type", "yaml", "`yaml` or a json")
+	flag.Parse()
+	validateFlags(action, inputFileName, inputFileType)
+	if inputFileType == "json" {
+		fmt.Println("Error: json file type is not supported.")
+		os.Exit(5)
+	}
+	errs := []string{}
+	if inputFileType == "yaml" {
+		objs, err := readFile(inputFileName)
+		if err != nil {
+			fmt.Printf("Error while reading input spec: %v\n", err)
+			os.Exit(2)
+		}
+		for _, o := range objs {
+			marshalledBytes, err := o.MarshalJSON()
+			if err != nil {
+				fmt.Printf("Error while marshalling json: %v\n", err)
+				os.Exit(3)
+			}
+			err = nil
+			switch kind := o.GetKind(); kind {
+			case "Task":
+				// No validation.
```

Comment:
Could you also please add the validate for the task as well? as it's the case exist. Thanks

Reply:
This `Task` is validated by tektoncd, and it cannot contain a custom task definition. So, I have not added a validation for it.